src/receive_imf.rs

```diff
@@ -581,13 +581,15 @@ async fn add_parts(
             {
                 chat_id = Some(new_chat_id);
                 chat_id_blocked = new_chat_id_blocked;
+            }
+        }
 
-                // if the chat is somehow blocked but we want to create a non-blocked chat,
-                // unblock the chat
-                if chat_id_blocked != Blocked::Not && create_blocked == Blocked::Not {
-                    new_chat_id.unblock(context).await?;
-                    chat_id_blocked = Blocked::Not;
-                }
+        // if the chat is somehow blocked but we want to create a non-blocked chat,
+        // unblock the chat
+        if chat_id_blocked != Blocked::Not && create_blocked != Blocked::Yes {
+            if let Some(chat_id) = chat_id {
+                chat_id.set_blocked(context, create_blocked).await?;
```

Comment:
Here, we can discuss about this:

When we have a blocked group, and a contact that's neither blocked nor accepted (i.e. a "Request"-Contact) sends a message into it.

Should the group stay blocked, or should the group become a "Request"?

I think the group should become a "Request" because we currently don't allow blocking of groups via the UI, so AFAIK the only reason for a group to be blocked is because the contact who sent the first message into it is blocked. In this case, it would be surprising if this prevented the group from appearing as a "Request".

What do y'all think, esp. @adbenitez who probably has the most experience with blocking spammers?

Reply:
@r10s sure have opinions on this, I read from him that the groups are blocked to avoid them re-appearing if you leave a group and delete it, in such situation only adding you again should create the contact request, 

yes, this is quite controversial, good and bad at the same time, ideally this should be undo and allow properly blocking and unblocking groups, but that never landed so here we are

Reply:
my opinion: I agree with Hocuri, don't block groups until properly implemented

Reply:
> I read from him that the groups are blocked to avoid them re-appearing if you leave a group and delete it

If you leave a group, it doesn't become blocked, but added to the `leftgrps` sql table. This isn't touched by this PR here.

> ideally this should be undo and allow properly blocking and unblocking groups

Just so we are on the same page: I think that the main thing left to do here is thinking about and implementing an unblocking UI?

Reply:
I thinks it intersects with https://github.com/deltachat/deltachat-core-rust/pull/4185. Generally speaking, there's no need to block groups, but contacts. A contact is the unit of trust. So, if a blocked contact adds us to a group, just ignore. If it's an unknown contact, make it a Request apparently. But if we missed some messages (because of delivery problems f.e.), there may be a situation when we have a message containing us in To, but we don't have a referenced one by InReplyTo, so we don't know who added us. Then we can make it a Request if there are any non-blocked contacts in the group.

Reply:
I think it's OK to keep things simple, I can't imagine complaining a user that they were added by some unblocked contact but didn't get this message but then a blocked contact wrote into the group and now the group should be shown because they were actually added by the unblocked contact.

Anyway, you approved so seem to be about happy as-is, and if @r10s objects which I kinda don't expect then we can always do a follow-up PR.

Reply:
Yes, as for this PR, it looks good to me. What i wrote above is for future work. Won't things be simpler if we get rid of blocked groups at all? So, if an accpeted contact adds us, we join the group. If it's a new contact, then make it Request. But if it's a blocked contact, just don't bother the user.

Reply:
As for handling of missing additions/removals and orphaned InReplyTo-s, there's https://github.com/deltachat/deltachat-core-rust/pull/4185 already that anyway should handle all these things, i don't think that removing blocking groups comlicates it